pkg/api/handlers/compat/containers.go

```diff
@@ -661,3 +662,145 @@ func RenameContainer(w http.ResponseWriter, r *http.Request) {
+	// the container's own resources
+	resources := new(spec.LinuxResources)
+	oldResources := ctr.LinuxResources()
+	if oldResources != nil {
+		if err := libpod.JSONDeepCopy(oldResources, resources); err != nil {
+			utils.Error(w, http.StatusInternalServerError, fmt.Errorf("copying old resource limits: %w", err))
+			return
+		}
+	}
+
+	// CPU limits
+	cpu := resources.CPU
+	if cpu == nil {
+		cpu = new(spec.LinuxCPU)
+	}
+	useCPU := false
+	if options.CPUShares != 0 {
+		shares := uint64(options.CPUShares)
+		cpu.Shares = &shares
+		useCPU = true
+	}
+	if options.CPUPeriod != 0 {
+		period := uint64(options.CPUPeriod)
+		cpu.Period = &period
+		useCPU = true
+	}
+	if options.CPUQuota != 0 {
+		cpu.Quota = &options.CPUQuota
+		useCPU = true
+	}
+	if options.CPURealtimeRuntime != 0 {
+		cpu.RealtimeRuntime = &options.CPURealtimeRuntime
+		useCPU = true
+	}
+	if options.CPURealtimePeriod != 0 {
+		period := uint64(options.CPURealtimePeriod)
+		cpu.RealtimePeriod = &period
+		useCPU = true
+	}
+	if options.CpusetCpus != "" {
+		cpu.Cpus = options.CpusetCpus
+		useCPU = true
+	}
+	if options.CpusetMems != "" {
+		cpu.Mems = options.CpusetMems
+		useCPU = true
+	}
+	if useCPU {
+		resources.CPU = cpu
+	}
+
+	// Memory limits
+	mem := resources.Memory
+	if mem == nil {
+		mem = new(spec.LinuxMemory)
+	}
+	useMem := false
+	if options.Memory != 0 {
+		mem.Limit = &options.Memory
+		useMem = true
+	}
+	if options.MemorySwap != 0 {
+		mem.Swap = &options.MemorySwap
+		useMem = true
+	}
+	if options.MemoryReservation != 0 {
+		mem.Reservation = &options.MemoryReservation
+		useMem = true
+	}
+	if useMem {
+		resources.Memory = mem
+	}
+
+	// PIDs limit
+	if options.PidsLimit != nil {
+		if resources.Pids == nil {
+			resources.Pids = new(spec.LinuxPids)
+		}
+		resources.Pids.Limit = *options.PidsLimit
+	}
+
+	// Blkio Weight
+	if options.BlkioWeight != 0 {
+		if resources.BlockIO == nil {
+			resources.BlockIO = new(spec.LinuxBlockIO)
+		}
+		resources.BlockIO.Weight = &options.BlkioWeight
+	}
+
+	// Restart policy
+	var restartPolicy *string
+	var restartRetries *uint
+	if options.RestartPolicy.Name != "" {
+		localPolicy := string(options.RestartPolicy.Name)
+		restartPolicy = &localPolicy
+		if localPolicy == "on-failure" {
+			if options.RestartPolicy.MaximumRetryCount < 0 {
+				utils.Error(w, http.StatusInternalServerError, errors.New("restart retries must be positive or 0"))
+				return
+			}
+			localRetries := uint(options.RestartPolicy.MaximumRetryCount)
+			restartRetries = &localRetries
+		} else if options.RestartPolicy.MaximumRetryCount != 0 {
+			utils.Error(w, http.StatusInternalServerError, errors.New("cannot set restart retries unless policy is on-failure"))
+			return
+		}
+	} else if options.RestartPolicy.MaximumRetryCount != 0 {
+		utils.Error(w, http.StatusInternalServerError, errors.New("cannot set restart retries without setting restart policy"))
+		return
+	}
```

Comment:
this is already validated in the backend code, so I think we should drop the validations.